### ISSUE TITLE
Inspect variants of DNN neuron model

### ISSUE THREAD

**Issue by dbeinhauer** [dbeinhauer/mcs-source#2]
Inspect various parameters of shared DNN model of neuron.

- [x] #3
- [x] #4
- [x] #5
- [x] #13

**Comment by dbeinhauer** [comment]
After throughout experiments it seems that general properties of the model are the following:

- usage of complex module improves the model performance
- usage of LSTM module is better than feed-forward DNNs
- ideal layer size is 10
- ideal number of layers is 5 for DNN and 1 for RNNs (no hidden time step)
- residual connections significantly help in terms of model stability during training
- usage of custom final activation function (LeakyTanh) that is reasonably bounded ([0, 5]) significantly helps in terms of model stability and final evaluation performance
- usage of hidden time steps in the outer RNN model (connections between the layers) worsen the performance and should not be used
- gradient clipping is not necessary and in fact sometimes complicates the training (since usage of custom activation function it is not needed anymore)
- dropout probably won't help in training
- layer normalization slightly improves the model performance (not batch norm though)
